### PR TITLE
Fix and improve build

### PR DESCRIPTION
**Pull request by nullableVoidPtr**
Currently, the source dist does not include the ANTLR files needed to regenerate the Python code, so I have amended the `package_data` configuration to fix the typo and ensured it was included.

After that, I moved everything I could to the new pyproject standard and ensured things like the wheel and source dist were mostly the same; the only difference was things like METADATA entries, as seen with `diffoscope`:
```
│ ├── python-fx-0.1.1/src/python_fx.egg-info/PKG-INFO
│ │ @@ -1,15 +1,14 @@
│ │  Metadata-Version: 2.1
│ │  Name: python-fx
│ │  Version: 0.1.1
│ │  Summary: A python-native fx-alike terminal JSON viewer.
│ │ -Home-page: https://github.com/cielong/pyfx
│ │ -Author: Yutian Wu
│ │ -Author-email: yutianwu@umich.edu
│ │ +Author-email: Yutian Wu <yutianwu@umich.edu>
│ │  License: MIT
│ │ +Project-URL: repository, https://github.com/cielong/pyfx
│ │  Keywords: fx,pyfx,json viewer,tui
│ │  Classifier: Development Status :: 4 - Beta
│ │  Classifier: Programming Language :: Python :: 3 :: Only
│ │  Classifier: Environment :: Console
│ │  Classifier: Operating System :: POSIX
│ │  Classifier: Operating System :: MacOS :: MacOS X
│ │  Classifier: License :: OSI Approved :: MIT License
```

Please let me know if you are confused about any aspect of the changes; I've only noticed these problems when I tried to package this repo for a downstream project, and ran into incompatible ANTLR runtime issues on Arch.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # content of: tox.ini , put in same dir as setup.py
 [tox]
-envlist = clean, style_check, py38, doctest
+envlist = clean, style_check, py310, doctest
 
 [gh-actions]
 python =
-    3.8: py38
+    3.10: py310
```
Comment (cielong) [inline tox.ini:7]: Thanks for this great change. Can you add py310 but not remove py38 though? This project is supposed to be support python version >= 3.8, I'm worried change this directly may not be able to guarantee this anymore.

**Comment by cielong** [comment]
FYI, the CI/CD fails with 

> ERROR: InterpreterNotFound: python3.10

You probably also need to change the config files under .github/

File: tox.ini (hunk 1 above)
Comment (nullableVoidPtr) [reply tox.ini]: I have updated the tox environments and also the version matrix on the github workflows; not sure if I've done it correctly, so happy to correct this further.

File: tox.ini [hunk 2]
```diff
@@ -1,10 +1,11 @@
 # content of: tox.ini , put in same dir as setup.py
 [tox]
-envlist = clean, style_check, py38, doctest
+envlist = clean, style_check, py38, py310, doctest
```
Comment (cielong) [inline tox.ini:3]: https://tox.wiki/en/latest/config.html#conf-skip_missing_interpreters

probably something like.
```
skip_missing_interpreters = true
```

The reason why it fails seems to be due to the fact that the way we set up in github actions has only single python version installed. So we need to skip py310 test for actions with only python 3.8 installed.

File: .travis.yml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 language: python
 jobs:
   include:
-    - name: "Python 3.8.0 on Xenial Linux"
```
Comment (cielong) [inline .travis.yml:4]: Thanks for catching this. I think GitHub actions is good enough, I probably will delete this config later.

**Pull request merged**